src/osd/scrubber/pg_scrubber.cc

```diff
@@ -2426,45 +2524,125 @@ ostream& PgScrubber::show(ostream& out) const
   return out << " [ " << m_pg_id << ": " << m_flags << " ] ";
 }
 
-int PgScrubber::asok_debug(std::string_view cmd,
-			   std::string param,
-			   Formatter* f,
-			   stringstream& ss)
+int PgScrubber::asok_debug(
+    Formatter* f,
+    std::string_view prefix,
+    std::string_view cmd,
+    std::string_view param,
+    std::string_view arg1)
 {
-  dout(10) << __func__ << " cmd: " << cmd << " param: " << param << dendl;
+  dout(10) << fmt::format(
+		  "asok_debug: pr={} cmd={}, param={}, val={}", prefix, cmd,
+		  param, arg1)
+	   << dendl;
 
-  if (cmd == "block") {
-    // 'm_debug_blockrange' causes the next 'select_range' to report a blocked
-    // object
-    m_debug_blockrange = 10;  // >1, so that will trigger fast state reports
+  auto set_on = prefix == "scrubdebug"sv || prefix == "scrubdebug_set"sv;
+  int ret{0};
 
-  } else if (cmd == "unblock") {
-    // send an 'unblock' event, as if a blocked range was freed
+  /// \todo (re)consider using a hash-based dispatch table here
+  if (cmd == "block"sv) {
```

Comment:
Not specifically an issue of performance (i would guess that the compiler handles that - but I'll check, out of curiosity). It's just that cmd is a stringview, and it's "more correct" to compare it to a sv.

Reply:
@rzarzynski - I did check just now. See for example https://godbolt.org/z/4rz78Ezdb

The same code is produced in both cases. I just preferred to have the string type visible. Not really important.

Reply:
Well, as expected – the compiler does a good job :-) (abstracting from the relevance to this particular appliance).

> It's just that cmd is a stringview, and it's "more correct" to compare it to a sv.

I know it's very little cost but I'm still dubious that spending two extra chars brings any benefit; actually, it already confused at least one reader and, I guess, it will confuse more over code's life time.

Reply:
I have removed the _sv in that function